File: Data Structures & Algorithms/longest-palindromic-substring/submission-1.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        res = ""
        reslen = 0
        n = len(s)
        for i in range(n):
            for j in range(i, n):
                l, r = i, j
                while l<r and s[l] == s[r]:
                    l+=1
                    r-=1
                if l>=r and reslen<(j-i+1):
                    res = s[i:j+1]
                    reslen = (j-i+1)
        return res

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-1.java
class Solution {
    public int evalRPN(String[] tokens) {
        Stack<String> stack = new Stack<>();
        int res = 0;
        for(int i=0;i<tokens.length;i++)
        {
            if("+-*/".contains(tokens[i]))
            {
                int b = Integer.parseInt(stack.pop());
                int a = Integer.parseInt(stack.pop());
                if(tokens[i].equals("+"))
                {
                    res=a+b;
                }
                else if(tokens[i].equals("-"))
                {
                    res=a-b;
                }
                else if(tokens[i].equals("*"))
                {
                    res=a*b;
                }
                else if(tokens[i].equals("/"))
                {
                    res=a/b;
                }
                stack.push(String.valueOf(res));
            }
            else
            {
                stack.push(tokens[i]);
            }
        }
        return Integer.parseInt(stack.pop());
    }
}
